### PR TITLE
Deserialize JSON directly to SidelinePayload

### LINKED ISSUE #54: retrieveSidelineRequest() should deserialize directly to SidelinePayload
For reference see:
https://github.com/salesforce/storm-dynamic-spout/pull/48/files#diff-2023ea74ef0624d2e7668f3a80d170ecR167

Now that we've switched to Gson we can deserialize direct to an entity (yey). Additionally it might make sense to add a method like CuratorHelper.readJson(String path, Class clazz) to make this even more seamless.

### PR DESCRIPTION
**Pull request by stanlemon**
Resolves #54 

This ended up being a bit tricky thanks to the way that we serialize `FilterChainStep` instances.

### CODE REVIEW

File: src/test/java/com/salesforce/storm/spout/sideline/persistence/ZookeeperPersistenceAdapterTest.java [hunk 1]
```diff
@@ -338,7 +357,27 @@ public void testEndToEndRequestStatePersistenceWithValidationWithIndependentZkCl
         final String storedDataStr = new String(storedDataBytes, Charsets.UTF_8);
         logger.info("Stored data string {}", storedDataStr);
         assertNotNull("Stored data string should be non-null", storedDataStr);
-        assertEquals("Got unexpected state", expectedStoredState, storedDataStr);
+        Map expectedMap = gson.fromJson(expectedStoredState, HashMap.class);
+
+        System.out.println(expectedStoredState);
```
Comment (Crim) [inline src/test/java/com/salesforce/storm/spout/sideline/persistence/ZookeeperPersistenceAdapterTest.java]: Probably replace these with logger statements?

Comment (stanlemon) [reply src/test/java/com/salesforce/storm/spout/sideline/persistence/ZookeeperPersistenceAdapterTest.java]: Whoops, good catch.

Comment (stanlemon) [reply src/test/java/com/salesforce/storm/spout/sideline/persistence/ZookeeperPersistenceAdapterTest.java]: Dropped altogether during the rebase.

File: src/main/java/com/salesforce/storm/spout/dynamic/filter/StaticMessageFilter.java [hunk 2]
```diff
@@ -38,10 +38,9 @@
     /**
      * We need a way to make this instance unique from others, so we use a UUID.
      */
-    private final String id;
+    private String id = UUID.randomUUID().toString();
```
Comment (Crim) [inline src/main/java/com/salesforce/storm/spout/dynamic/filter/StaticMessageFilter.java]: why not final?

Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/filter/StaticMessageFilter.java]: Something weird was happening when it deserialized when I set it inside of the constructor, I think it has to do with the way GSON reflects, but I'm not sure honestly. This worked around it.  This reminds me... this class should be moved into src/test.

**Pull request merged**